extension/pgtelemetry--1.0--1.2.sql

```diff
@@ -0,0 +1,55 @@
+-- version compatibility
+do $d$ begin
+if current_setting('server_version_num')::int < 100000 then
+   CREATE FUNCTION pg_current_xlog_location() RETURNS pg_lsn language sql as $$
+       select pg_current_wal_lsn();
+   $$;
+   CREATE FUNCTION pg_last_xlog_replay_location() returns pg_lsn language sql as $$
+       select pg_last_wal_replay_lsn();
+   $$;
+   -- filter out all non-backend connections
+   create view client_stat_activity as select * from pg_stat_activity where backend_type = 'client backend';
+else
+  -- in pre PostgreSQL 10 we only need to filter out autovacuum workers;
+  -- however, there is no backend_type field, so we had to rely on the query text.
+   create view client_stat_activity as select * from pg_stat_activity where query not like 'autovacuum:%';
+end if;
+end;$d$ language plpgsql;
+
+
+create or replace view lock_queue_size_five_minutes_wait as
+select count(1) from pg_locks join client_stat_activity using(pid)
+where granted = 'f' and extract('epoch' from now() - query_start) > 300;
```

Comment:
I wouldn't use view in this case at all; instead I'd put that query to the prometheus exporter or any other tool which will collect the metrics

Reply:
or maybe wrap that query into a function with the interval as an argument

Reply:
Well, I was not sure what is the line between creating a view in the pgtelemetry and putting the bare query into the Postgres exporter, and I thought the view would make it easier to reuse the query (i.e. run it manually), as well as provide better visibility into the kind of queries we use for monitoring.

Reply:
changing of that interval value might be expensive, as it will lead to a new version of pgtelemetry

Reply:
Right, I am on board with making it a function with the interval parameter, although I don't think we are going to change it that often (on the other hand, you never know... :-))